addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/service/NetworkService.java

```diff
@@ -110,35 +112,78 @@ public void setUseSystemPing(boolean useSystemPing) {
         this.useSystemPing = useSystemPing;
     }
 
-    public void startAutomaticRefresh(ScheduledExecutorService scheduledExecutorService,
-            final StateUpdate stateUpdate) {
-        Runnable runnable = new Runnable() {
+    /**
+     * Create a runner to be called by a scheduler periodically. The runner
+     * will update the device state and inform the callback listener.
+     *
+     * @return Return a runner.
+     */
+    private Runnable createUpdateListenerRunnable() {
+        return new Runnable() {
             @Override
             public void run() {
                 try {
-                    stateUpdate.newState(updateDeviceState());
+                    updateListener.updatedDeviceState(updateDeviceState());
                 } catch (InvalidConfigurationException e) {
-                    stateUpdate.invalidConfig();
+                    updateListener.invalidConfig();
                 }
             }
         };
+    }
 
-        refreshJob = scheduledExecutorService.scheduleAtFixedRate(runnable, 0, refreshInterval, TimeUnit.MILLISECONDS);
-
-        if (dhcplisten) {
-            try {
-                ReceiveDHCPRequestPackets.register(InetAddress.getByName(hostname).getHostAddress(), stateUpdate);
-            } catch (SocketException | UnknownHostException e) {
-                logger.error("Cannot use DHCP listen: {}", e.getMessage());
+    /**
+     * Cancel a running refreshJob. Will not throw if no job is running
+     * or the job is already canceled.
+     */
+    private void cancelRefreshJob() {
+        try {
+            if (refreshJob != null) {
+                refreshJob.cancel(true);
             }
+        } catch (Exception ignore) {
         }
     }
 
+    /**
+     * Start/Restart a fixed scheduled runner to update the devices reach-ability state.
+     *
+     * @param scheduledExecutorService A scheduler to run pings periodically.
+     * @param stateChangeListener Your callback listener
+     */
+    public void startAutomaticRefresh(ScheduledExecutorService scheduledExecutorService) {
+        cancelRefreshJob();
+        refreshJob = scheduledExecutorService.scheduleAtFixedRate(createUpdateListenerRunnable(), 0, refreshInterval,
```

Comment:
I will redesign the code to not touch the scheduled job. The cache will be moved from the handler to the NetworkService.java class, and will respond to gui issued requests as well as scheduled jobs. This way my initial goal of not flooding devices with ping requests can be accomplished as well.